app/src/features/Inputs/StaticInput.tsx

```diff
@@ -145,6 +145,15 @@ const StaticInput = ({ input }: StaticInputProps): JSX.Element => {
     }
   };
 
+  const isDisabled = () => {
+    if (selectedNode) {
+      const fixedValueKey = Object.keys(
+        selectedNode?.definitionNode.definition
+      ).find((objectKey) => objectKey.includes("fixed"));
+      return fixedValueKey ? true : false;
+    }
+  };
```

Comment:
First of all I'd compute the `isDisabled` boolean value with a useMemo hook to prevent recalculation of its value on each render.

Then, it seems like this would work (logic speaking) because most of all if there is a staticInput created for the selectedNode, the node MUST be of `kind` primitive. But I'm sort of hesitating on the fact that maybe we should check on the staticValue and disabling it if it is the same as the fixed value..
Maybe @tevariou you'd have a suggestion ?

Reply:
not really but the second option makes sense